rir/src/compiler/debugging/stream_logger.cpp

```diff
@@ -0,0 +1,124 @@
+#include "stream_logger.h"
+#include "../../utils/Pool.h"
+#include <iomanip>
+
+namespace rir {
+namespace pir {
+
+void StreamLogger::startLogging(rir::Function* function) {
+    if (options.intersects(PrintDebugPasses)) {
+        if (!streams.count(function)) {
+            streams.insert(std::pair<rir::Function*, std::stringstream*>(
```

Comment:
`streams.emplace(function, new std::stringstream())` is the idiomatic way of inserting into a map. It's shorter, clearer, and constructs the pair directly in the map.

A slightly more verbose way is to use initializer lists: `streams.insert({function, new std::stringstream()})`, but I think it's messier. (Apparently, there used to an issue with move-only semantics.)

Calling the `std::pair` constructor or `std::make_pair` is more verbose, and can actually lead to unnecessary copies, because the map actually requires the type `std::pair<const rir::Function*, std::stringstream*>`.

See:
https://stackoverflow.com/a/19942856
https://stackoverflow.com/a/17174245